### ISSUE TITLE
Ecto Guide:  Generated Phoenix Model

### ISSUE THREAD

**Issue by mneumark** [phoenixframework/phoenix_guides#530]
The generated Phoenix model on line 207 no longer generates:
```elixir
  @required_fields ~w(name email bio number_of_pets)
  @optional_fields ~w()
```

Also, the default changeset is different:
```elixir
def changeset(model, params \\ :empty) do
  model
  |> cast(params, @required_fields, @optional_fields)
end

# vs

def changeset(struct, params \\ %{}) do
  struct
  |> cast(params, [:name, :email, :bio, :number_of_pets])
  |> validate_required([:name, :email, :bio, :number_of_pets])
end
```
- `model` is now `struct`
- `:empty` is now `%{}`
- `cast/4` is now a call to `cast/3` in addition to a new `validate_required/1`.

It seems that you can modify required vs. optional simply by altering the changeset, but I suspect that is not correct and I just don't have an example to model optionals after given the example in the guide.

**Comment by ybur-yug** [comment]
@mneumark I wrote up a PR that should have this handled if you'd like to proofread (linked above)

**Comment by ybur-yug** [comment]
This can be closed and was addressed in #534 
(CC @Gazler or whomever would be good to close)
